include/device.h

```diff
@@ -333,7 +333,7 @@ typedef int16_t device_handle_t;
 #define DEVICE_DECLARE(name) static const struct device DEVICE_NAME_GET(name)
 
 typedef void (*device_pm_cb)(const struct device *dev,
-			     int status, void *context, void *arg);
+			     int status, uint32_t *state, void *arg);
```

Comment:
Maybe use enum instead?

Reply:
I intend to do it, but I will need to review the states and unify with fsm_state. There is another pr doing a big cleanup and will be better to have that one before doing this type change. Here is just change the signature since all places were using it was doing cast o uint32_t